lib/src/call/quality_issue.dart

```diff
@@ -0,0 +1,398 @@
+/// if the issue is not detected or resolved.
+///
+/// If no audio receive is detected on several remote audio streams,the
+/// event will be invoked for each of the remote audio streams with the issue.
+///
+/// If the issue level is `VIQualityIssueLevel.VIQualityIssueLevelCritical`
+/// the event will not be invoked with the level VIQualityIssueLevelNone in cases:
+///
+/// * The (conference) call ended
+/// * The endpoint left the conference call -
+/// [VIEndpoint.onEndpointRemoved] is invoked
+///
+/// The issue is not detected for the following cases:
+///
+/// * The endpoint put the call on hold via [VICall.hold]
+/// * The endpoint stopped sending audio during a call via [VICall.sendAudio]
+///
+/// Possible reasons:
+///
+/// * Poor internet connection on the client or the endpoint
+/// * Connection lost on the endpoint
+class NoAudioReceive implements QualityIssueEvent {
+  /// Id remote audio stream the issue occured on.
+  final String audiostreamId;
+
+  /// Id endpoint the issue belongs to.
+  final String endpointId;
+
+  /// Issue level
+  final VIQualityIssueLevel level;
+
+  NoAudioReceive({
+    required this.audiostreamId,
+    required this.endpointId,
+    required this.level,
+  });
+}
+
+/// Represents class reporting that no video is received on the
+/// remote video stream.
+///
+/// Issue level can be only `VIQualityIssueLevel.VIQualityIssueLevelCritical`
+/// if the issue is detected or `VIQualityIssueLevel.VIQualityIssueLevelNone`
+/// if the issue is not detected or resolved.
+///
+/// If no video receive is detected on several remote video streams,the
+/// event will be invoked for each of the remote video streams with the issue.
+///
+/// If the issue level is `VIQualityIssueLevel.VIQualityIssueLevelCritical`
+/// the event will not be invoked with the level VIQualityIssueLevelNone in cases:
+///
+/// * The (conference) call ended
+/// * The remote video stream was removed -
+/// [VIEndpoint.onRemoteVideoStreamRemoved] is invoked
+/// * The endpoint left the conference call -
+/// [VIEndpoint.onEndpointRemoved] is invoked
+///
+/// The issue is not detected for the following cases:
+///
+/// * The endpoint put the call on hold via [VICall.hold]
+/// * The endpoint stopped sending audio during a call via [VICall.sendVideo]
+/// * Video receiving was stopped on the remote video stream via
+/// [VIEndpoint.stopReceiving]
+///
+/// Possible reasons:
+///
+/// * Poor internet connection on the client or the endpoint
+/// * Connection lost on the endpoint
+/// * The endpoint's application has been moved to the background state on an
+/// iOS device (camera usage is prohibited while in the background on iOS)
+class NoVideoReceive implements QualityIssueEvent {
+  /// Id remote video stream the issue occured on.
+  final String videostreamId;
+
+  /// Id endpoint the issue belongs to.
+  final String endpointId;
+
+  /// Issue level
+  final VIQualityIssueLevel level;
+
+  NoVideoReceive({
```

Comment:
probably it would be better to replace this constructor with `_fromMap` named constructor and parse the data in it

Reply:
done